### ISSUE TITLE
Excessive reads on Redis

### ISSUE THREAD

**Issue by LarsVQ** [TykTechnologies/tyk-pump#752]
Hello 👋 

We have deployed a Tyk gateway that is processing about 200 to 450 requests/sec and we expose Prometheus metrics via Tyk Pump.
Currently we are facing an issue where it seems like the pump is reading excessive amounts of data from Redis.
As you can see in Redis metrics in the screenshot below, Tyk is only sending about 70MiB to Redis while at the same time consuming more than 6GiB.

<img width="1728" alt="Screenshot 2023-12-04 at 17 40 58" src="https://github.com/TykTechnologies/tyk-pump/assets/152897224/90bae946-94f6-4528-ae94-3e234696cce7">

The configuration for the gateway look like this (some parts omitted):
```json
{
    "storage": {
        "type": "redis",
        "addrs": [ "HOST" ],
        "username": "USERNAME",
        "password": "PASSWORD",
        "use_ssl": true,
        "database": 0,
        "optimisation_max_idle": 2000,
        "optimisation_max_active": 4000
    },
    "enable_analytics": true,
    "analytics_config": {
        "purge_delay": -1,
        "ignored_ips": []
    },
    "optimisations_use_async_session_write": true,
    "enable_non_transactional_rate_limiter": true,
    "enable_sentinel_rate_limiter": false,
    "enable_redis_rolling_limiter": false,
    "allow_master_keys": false,
    "hash_keys": true,
    "close_connections": false,
    "http_server_options": {
        "enable_websockets": true
    },
    "global_session_lifetime": 100,
    "force_global_session_lifetime": true,
    "max_idle_connections_per_host": 500,
    "enable_jsvm": false,
}
```

The pump configuration looks like this (some parts omitted):
```json
{
    "analytics_storage_type": "redis",
    "analytics_storage_config": {
        "type": "redis",
        "host": "HOST",
        "port": "PORT",
        "username": "USERNAME",
        "password": "PASSWORD",
        "redis_use_ssl": true,
        "database": 0,
        "optimisation_max_idle": 200,
        "optimisation_max_active": 500
    },
    "enable_analytics": true,
    "analytics_config": {
        "purge_delay": -1,
        "ignored_ips": []
    },
    "purge_delay": 4,
    "dont_purge_uptime_data": true,
    "log_format": "json",
    "pumps": {
        "prometheus": {
        "type": "prometheus",
        "meta": {
            "listen_address": "0.0.0.0:9090",
            "path": "/metrics",
            "disabled_metrics": ["tyk_http_status_per_key", "tyk_latency"],
            "custom_metrics": [
            {
                "name": "tyk_http_status_per_api",
                "description": "HTTP status codes per API",
                "metric_type": "counter",
                "labels": [ "response_code","api_id","api_name","method" ]
            },
            {
                "name": "tyk_latency_per_api",
                "description": "Latency buckets per API",
                "metric_type": "histogram",
                "labels": [ "api_id","api_name","method" ],
                "buckets": [1,2,5,10,20,30,50,75,100,200,300,400,500,1000,2000,5000,10000,30000,60000]
            }
            ]
        }
        },
        "stdout": {
        "type": "stdout",
        "meta" : {
            "log_field_name": "tyk-analytics-record",
            "format": "json"
        }
        }
    }
}
```

Do you have any idea what could cause these extreme amounts of reads on Redis?
Any help would be much appreciated!

**Comment by tbuchaillot** [comment]
hey @LarsVQ ! Thanks for the information! 

Would you mind attaching some `pump logs` and redis monitor logs on those spikes? It would be really helpful to identify the issue!

**Comment by LarsVQ** [comment]
Hey @tbuchaillot, sorry for the late response.
So after investigation on our side we figured out that we misinterpreted the metrics provided by AWS, as they were showing an average over 5 minutes. So the actual network traffic is much lower, though we were still exceeding our baseline bandwidth and sometimes hit the burst limit on our instance.
So in the end we solved it by upscaling the instance to have a higher baseline bandwidth.
